# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmJS.cpp
# Commit: 6d6adcf77b5a
# Full Hash: 6d6adcf77b5a99316ef743323ccf362972f020dd
# Author: Yury Delendik <ydelendik@mozilla.com>
# Date: 2021-10-22 10:00:52
# Regressor Bug: 1545131
# File Overlap Count: 1
# Description:
#   Bug 1545131 - compress wasm alt data entries (r=jandem)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27992
# ==============================================================================

diff -r a6180f7f6296 -r 6d6adcf77b5a js/src/wasm/WasmJS.cpp
--- a/js/src/wasm/WasmJS.cpp	Thu Oct 21 21:54:52 2021 +0000
+++ b/js/src/wasm/WasmJS.cpp	Thu Oct 21 22:08:14 2021 +0000
@@ -752,11 +752,11 @@
   Bytes* serialized;
   explicit SerializeListener(Bytes* serialized) : serialized(serialized) {}
 
-  void storeOptimizedEncoding(JS::UniqueOptimizedEncodingBytes bytes) override {
+  void storeOptimizedEncoding(const uint8_t* bytes, size_t length) override {
     MOZ_ASSERT(!called);
     called = true;
-    if (serialized->resize(bytes->length())) {
-      memcpy(serialized->begin(), bytes->begin(), bytes->length());
+    if (serialized->resizeUninitialized(length)) {
+      memcpy(serialized->begin(), bytes, length);
     }
   }
 };